objects/src/transaction/inputs.rs

```diff
@@ -327,64 +326,95 @@ fn build_nullifier_commitment<T: ToNullifier>(notes: &[T]) -> Digest {
 // INPUT NOTE
 // ================================================================================================
 
+const AUTHENTICATED: u8 = 0;
+const UNVERIFIED: u8 = 1;
+
 /// An input note for a transaction.
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct InputNote {
-    note: Note,
-    proof: NoteInclusionProof,
+pub enum InputNote {
+    /// Input notes which have an authentication path and are authenticated by the transaction
+    /// kernel.
```

Comment:
nit: I'd probably explain what "authentication" means in this context. Maybe something like: "Input notes whose existences in the chain is verified by the transaction kernel."

Reply:
changed